### PR TITLE
Move geometry and KokkosExt related header files to separate directories

### PR DESCRIPTION
**Pull request by aprokop**

### CODE REVIEW

**Review by dalg24 — COMMENTED** [review]
I actually regret that we install a raw `details` directory directly into `<prefix>/include`
So please `geometry -> arborx/geomety` and `kokkos_ext -> arborx/kokkos_ext`
**edit** Not necessarily in our directory structure, but definitely in what we install.

**Comment by aprokop** [comment]
@dalg24 I changed installation of all headers into `include/ArborX`.

**Comment by aprokop** [comment]
@dalg24 This passes all our tests, including installation. Anything else you want to do here, or can it be merged?

**Review by dalg24 — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -115,14 +112,14 @@ install(FILES
   DESTINATION lib/cmake/ArborX )
 
 if(ARBORX_ENABLE_MPI)
-  install(DIRECTORY ${PROJECT_SOURCE_DIR}/src/ DESTINATION include
+  install(DIRECTORY ${PROJECT_SOURCE_DIR}/src/ DESTINATION include/ArborX
```
[the remark this answers is not in the document]
Comment (aprokop) [reply CMakeLists.txt:118]: I'm not sure I understand. The CI installation tests pass.

[the remark this answers is not in the document]
Comment (aprokop) [reply CMakeLists.txt:118]: Comment was outdated, prior to fixing the issue.

**Pull request merged**